NSFS | S3 | Versioning: Double delete of object in version-enabled bucket fails

### Environment info
- NooBaa Version: ** 5.15.4**
- Platform: **CentOS 8, deployed using [https://github.com/IBM/StorageScaleVagrant](https://github.com/IBM/StorageScaleVagrant)**

### Actual behavior

When performing double deletes (without versionId) on objects in version-enabled buckets, the second delete fails with
```
E           botocore.errorfactory.NoSuchKey: An error occurred (NoSuchKey) when calling the DeleteObject operation: The specified key does not exist.
```

### Expected behavior

The delete should pass due to the AWS idempotency principle. An additional DeleteMarker should get added.

### Steps to reproduce

Execute the following addition to Ceph s3-tests:
```
        client = get_client()
        bucket_name = get_new_bucket(client=client)
    
        # The following operation uses another client instance -> following operations will fail
    #    check_configure_versioning_retry(bucket_name, "Enabled", "Enabled")
        _ces_safe_check_configure_versioning_retry(client, bucket_name, "Enabled", "Enabled")
    
        key_names = ['bar', 'baz', 'foo']
        i = 1
        for key in key_names:
            response = client.put_object(Bucket=bucket_name, Key=key, Body='content-{}'.format(i))
            version_id = response['VersionId']
            print('Key {} version_id {}'.format(key, version_id))
            i = i + 1
    
        # 1st pass delete
        for key in key_names:
            response = client.delete_object(Bucket=bucket_name, Key=key)
    
       # 2nd pass delete
        for key in key_names:
           response = client.delete_object(Bucket=bucket_name, Key=key)

       # 3rd pass delete
       for key in key_names:
            response = client.delete_object(Bucket=bucket_name, Key=key)

       response = client.list_object_versions(Bucket=bucket_name)
       objs_list = response['Versions']
       print(objs_list)
       assert len(response['DeleteMarkers']) == 9
```

### More information - Screenshots / Logs / Other output

Logs with "all" log level:
[noobaa_20240911_1.log.gz](https://github.com/user-attachments/files/16960819/noobaa_20240911_1.log.gz)

--- a/src/sdk/namespace_fs.js
+++ b/src/sdk/namespace_fs.js
@@ -2911,13 +2911,13 @@ class NamespaceFS {
                 const versioned_path = latest_ver_info && this._get_version_path(params.key, latest_ver_info.version_id_str);
                 const versioned_info = latest_ver_info && await this._get_version_info(fs_context, versioned_path);
 
-                gpfs_options = is_gpfs ?
+                dbg.log1('Namespace_fs._delete_latest_version:', latest_ver_info, versioned_path, versioned_info);
+                if (latest_ver_info) {
+                    gpfs_options = is_gpfs ?
                     await this._open_files_gpfs(fs_context, latest_ver_path,
                         undefined, undefined, undefined, undefined, true, versioned_info) :
                     undefined;
 
-                dbg.log1('Namespace_fs._delete_latest_version:', latest_ver_info, versioned_path, versioned_info, gpfs_options);
-                if (latest_ver_info) {
                     const suspended_and_latest_is_not_null = this._is_versioning_suspended() &&
                         latest_ver_info.version_id_str !== NULL_VERSION_ID;
                     const bucket_tmp_dir_path = this.get_bucket_tmpdir_full_path();

